scipy/special/_basic.py

```diff
@@ -2728,12 +2728,9 @@ def comb(N, k, *, exact=False, repetition=False):
         if int(N) == N and int(k) == k:
             # _comb_int casts inputs to integers, which is safe & intended here
             return _comb_int(N, k)
-        # otherwise, we disregard `exact=True`; it makes no sense for
-        # non-integral arguments
-        msg = ("`exact=True` is deprecated for non-integer `N` and `k` and will raise "
-               "an error in SciPy 1.16.0")
-        warnings.warn(msg, DeprecationWarning, stacklevel=2)
-        return comb(N, k)
+        else:
+            raise ValueError("Non-integer `N` and `k` with `exact=True` is not "
+                             "supported.")
```

Comment:
Going forward, please avoid redundant `else:` branches (apparently there isn't any active style enforcement for that). What I mean is that - generally speaking - the following pattern
```python
if cond:
    return ...
# happens unconditionally if cond is not true
```
In this particular case though, it'd actually be cleaner to do
```python
if not cond:
    raise ValueError()
return ...
```